internal/ui/backup/rate_estimator_test.go

```diff
@@ -0,0 +1,233 @@
+package backup
+
+import (
+	"math"
+	"testing"
+	"time"
+)
+
+const float64EqualityThreshold = 1e-6
+
+func almostEqual(a, b float64) bool {
+	if math.IsNaN(a) || math.IsNaN(b) {
+		panic("almostEqual passed a NaN")
+	}
+	return math.Abs(a-b) <= float64EqualityThreshold
+}
+
+func TestEstimatorDefault(t *testing.T) {
+	var start time.Time
+	e := newRateEstimator(start)
+	r := e.rate(start)
+	if math.IsNaN(r) || r != 0 {
```

Comment:
Using `rtest.Assert` from the import `rtest "github.com/restic/restic/internal/test"` would make this a bit cleaner. Then you can just use `rtest.Assert(r == 0, ...)`